client/src/screens/Patient/Patient.js

```diff
@@ -912,436 +787,439 @@ export default function Patient() {
+              onClose={() => dispatch(togglePatientInfoDialog())}
             />
           )}
-          applyForm={() => togglePatientInfoDialog()}
-          cancelForm={() => togglePatientInfoDialog()}
+          applyForm={() => dispatch(togglePatientInfoDialog())}
+          cancelForm={() => dispatch(togglePatientInfoDialog())}
           hideActions
           size="lg"
         />
       )}
 
-      {!!showPatientHistoryDialog && (
+      {!!patientInfo.historyDialog && (
         <Dialog
-          open={showPatientHistoryDialog}
+          open={patientInfo.historyDialog}
           title="Patient History"
           message={(
             <PatientHistoryDetails
               data={patientHistory}
-              onClose={togglePatientHistoryDialog}
+              onClose={() => dispatch(togglePatientHistoryDialog())}
             />
           )}
-          applyForm={() => togglePatientHistoryDialog()}
-          cancelForm={() => togglePatientHistoryDialog()}
+          applyForm={() => dispatch(togglePatientHistoryDialog())}
+          cancelForm={() => dispatch(togglePatientHistoryDialog())}
           hideActions
           size="md"
         />
       )}
-      {!!showAdminHistoryDialog && (
+      {!!adminNotes.historyDialog && (
         <Dialog
-          open={showAdminHistoryDialog}
+          open={adminNotes.historyDialog}
           title="Admin Notes History"
           message={(
             <AdminNotesHistory
-              onClose={toggleAdminHistoryDialog}
+              onClose={() => dispatch(toggleAdminHistoryDialog())}
               data={adminNotesHistory}
             // onLoad={() => fetchPatientHistory()}
             />
           )}
-          applyForm={() => toggleAdminHistoryDialog()}
-          cancelForm={() => toggleAdminHistoryDialog()}
+          applyForm={() => dispatch(toggleAdminHistoryDialog())}
+          cancelForm={() => dispatch(toggleAdminHistoryDialog())}
           hideActions
           size="md"
         />
       )}
 
-      {!!showFormsExpandDialog && (
+      {!!forms.expandDialog && (
         <Dialog
-          open={showFormsExpandDialog}
+          open={forms.expandDialog}
           title={" "}
-          message={
-            <FormDetails data={forms} onClose={toggleFormsExpandDialog} />
-          }
-          applyForm={() => toggleFormsExpandDialog()}
-          cancelForm={() => toggleFormsExpandDialog()}
+          message={(
+            <FormDetails
+              data={patientForms}
+              onClose={() => dispatch(toggleFormsExpandDialog())}
+            />
+          )}
+          applyForm={() => dispatch(toggleFormsExpandDialog())}
+          cancelForm={() => dispatch(toggleFormsExpandDialog())}
           hideActions
           size="md"
         />
       )}
 
-      {!!showFormsViewDialog && (
+      {!!forms.viewDialog && (
         <Dialog
-          open={showFormsViewDialog}
+          open={forms.viewDialog}
```

Comment:
I noticed that `Dialog` already has a boolean props `open`. Is it necessary to use `!!forms.viewDialog &&` condition as well? Just curious to know 😉

Reply:
Yes, there is a reason if we don't use "!!forms.viewDialog &&"  then the instance of the dialog remains in the DOM even when the dialog is not opened. It is just for the optimization of DOM!

Reply:
:+1: